ui.apps/src/main/content/jcr_root/apps/asset-share-commons/components/details/video/templates/options.html

```diff
@@ -0,0 +1,55 @@
+<!--/*
+  ~ Asset Share Commons
+  ~
+  ~ Copyright [2017]  Adobe
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+ */-->
+<template data-sly-template.image="${ @ image}">
+	<!--/* Img element with a max height */-->
+	<img class="ui centered image cmp-image" 
+		 src="${image.src}"
+         class="cmp-image--max-height"
+         style="height:${height @ context = 'styleToken'}px;"
+         data-sly-test.height="${properties['height']}"/>
+         
+	<!--/* Img element without a max height */-->
+	<img class="ui centered image cmp-image"
+		 src="${image.src}"
+		 data-sly-test="${!height}"/>
+</template>
+
+
+<template data-sly-template.video="${ @ video}">
+		<!--/* Video element with a max height */-->	
+        <video class="ui centered image cmp-image" class="cmp-image--max-height" poster="${properties['posterImage']}"
+               style="height:${height @ context = 'styleToken'}px;" controls controlsList="nodownload"
+               data-sly-test.height="${properties['height']}">
+         	<source src="${video.src}" 
```

Comment:
Should multiple <sources> point to the same URL?

What is the idea behind hardcoding the types in?

Could/should these somehow be auto-derived based on the available renditions?

Or is this just acting as a "blanket" set of URL -> Coded matching to help ensure whatever the selected rendition is, actually plays?

Reply:
We could do this way. 
<video src="video.mp4" controls></video>
However, the  best practice is to include a type attribute with <source> element that allows you to specify alternative video files which the browser may choose from. The browser will use the first recognized format.